lib/nucleus_rails/worker.rb

```diff
@@ -0,0 +1,62 @@
+require "active_job"
+
+module NucleusRails
+  class Worker < NucleusCore::Worker
+    def self.enabled?(worker_class)
+      Object.const_defined?(worker_class&.to_sym)
+    end
+
+    if enabled?(:Sidekiq)
+      class SidekiqAdapter < NucleusCore::Worker::Adapter
+        include Sidekiq
+
+        def perform(*args)
+          NucleusCore::Worker::Adapter.execute(*args)
+        end
+
+        def self.execute_async(class_name, method_name, args)
+          worker_args = args.slice!(*Sidekiq.options.keys)
+
+          set(worker_args).perform_async(class_name, method_name, args)
+        end
+      end
+    end
+
+    class ActiveJobAdapter < NucleusCore::Worker::Adapter
+      class ApplicationJob < ActiveJob::Base; end
+
+      class Implementation < ApplicationJob
```

Comment:
do we need an implementation class? Can we instantiate `ApplicationJob` and can `perform_later` in `execute_async`?